hackathon/views.py

```diff
@@ -109,37 +122,113 @@ def check_projects_scores(request, hack_id):
     and render final_score.html with the score table.
 
     """
-    score_categories = HackProjectScoreCategory.objects.all()
     hackathon = get_object_or_404(Hackathon, pk=hack_id)
-    judges = hackathon.judges.count()
-    number_categories = score_categories.count()
-    projects = list(HackTeam.objects.filter(
-        hackathon=hackathon).values_list('project', flat=True).distinct())
-
-    # Number of objects that should be in the HackProjectScore for each project
-    number_of_objects = judges * number_categories
-
-    for project in projects:
-        if HackProjectScore.objects.filter(project=project).count() != number_of_objects:
-            return render(request, 'hackathon/final_score.html', {"hackathon": hackathon.display_name})
-
-    # Calculate total score for each team
-    scores_list = []
-    for project in projects:
-        total_score = sum(list(HackProjectScore.objects.filter(
-            project=project).values_list("score", flat=True)))
-        scores_list.append(total_score)
+    HackAwardCategoryFormSet = modelformset_factory(
+                HackAwardCategory, fields=('id', 'display_name', 'winning_project'),
+                form=HackAwardCategoryForm, extra=0)
 
-    # Pair the teams with the total score
-    team_scores = {}
-    for i, score in enumerate(scores_list):
-        team_scores[str(HackProject.objects.get(id=projects[i]))] = score
-
-    # Sort the teams by score
-    sorted_teams_scores = sorted(
-        team_scores.items(), key=lambda x: x[1], reverse=True)
+    if request.method == 'POST':
+        hack_awards_formset = HackAwardCategoryFormSet(
+            request.POST,
+            queryset=HackAwardCategory.objects.filter(hackathon=hackathon))
+        if hack_awards_formset.is_valid():
+            hack_awards_formset.save()
+        else:
+            print(hack_awards_formset.errors)
+            pass
+        return redirect(reverse('hackathon:final_score', kwargs={'hack_id': hack_id}))
 
-    return render(request, 'hackathon/final_score.html', {'sorted_teams_scores': sorted_teams_scores, "hackathon": hackathon.display_name})
+    else:
+        team_scores = {}
+        category_scores_per_team = {}
+        score_categories = HackProjectScoreCategory.objects.all()
+        hackathon_projects = [team.project.id for team in hackathon.teams.all()
+                              if team.project]
+        scores = HackProjectScore.objects.filter(
+            project_id__in=hackathon_projects).all()
+        judges = hackathon.judges.all()
+
+        # Creating scores data structure
```

Comment:
Do we need this in here still? All commented-out for a reason (keeping for later?)

Reply:
This is just so other developers have a picture of what the data structure will look like.